File: fibonacci.c
#include <stdio.h>

void FiboSeq_Last(int last_num)
{
    int prev = 0;
    int curr = 1;
    int next = 1;

    printf("%i", prev);

    if (last_num != 0)
    {
        while (last_num >= curr)
        {
            printf(", %i", next);
            next = prev + curr;
            prev = curr;
            curr = next;
        }
    }
}

void FiboSeq(int fib_num)
{
    int prev = 0;
    int curr = 1;
    int next = 1;
    int i = 1;

    printf("%i", prev);

    if (fib_num != 0)
    {
        while (fib_num > i)
        {
            printf(", %i", next);
            next = prev + curr;
            prev = curr;
            curr = next;
            i++;
        }
    }
}

int main(void)
{
    int last_num;
    int fib_num;

    printf("Enter a number to be generated in a Fibonacci Sequence: ");
    scanf("%i", &last_num);
    printf("The Fibonacci Sequence up to %i is: ", last_num);
    FiboSeq_Last(last_num);
    printf("\nEnter how many numbers will be generated in a Fibonacci Sequence: ");
    scanf("%i", &fib_num);
    printf("The Fibonacci Sequence with %i numbers is: ", fib_num);
    FiboSeq(fib_num);
}

File: fibonacci.py
def FiboSeq_Last(last_num):
    prev = 0
    curr = 1
    next = 1
    fib = [prev]

    if (last_num != 0):
        while (last_num >= curr):
            fib.append(next)
            next = prev + curr
            prev = curr
            curr = next
        return fib

def FiboSeq(fib_num):
    prev = 0
    curr = 1
    next = 1
    i = 1
    fib = [prev]

    if (fib_num != 0):
        while (fib_num > i):
            fib.append(next)
            next = prev + curr
            prev = curr
            curr = next
            i = i + 1
        return fib

last_num = int(input("Enter a number to be generated in a Fibonacci Sequence: "))
print("The Fibonacci Sequence up to", last_num, " is:", ', '.join(map(str, FiboSeq_Last(last_num))))
fib_num = int(input("Enter how many numbers will be generated in a Fibonacci Sequence: "))
print("The Fibonacci Swquence with", fib_num, "numbers is:", ', '.join(map(str, FiboSeq(fib_num))))